test/powershell/Language/Operators/SplitOperator.Tests.ps1

```diff
@@ -0,0 +1,236 @@
+            $res = "a b c d" -split " ", 2
+            $res.count | Should Be 2
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b c d"
+
+            $res = "a b c d" -split " ", 0
+            $res.count | Should Be 4
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b"
+            $res[2] | Should Be "c"
+            $res[3] | Should Be "d"
+
+            $res = "a b c d" -split " ", -1
+            $res.count | Should Be 4
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b"
+            $res[2] | Should Be "c"
+            $res[3] | Should Be "d"
+        }
+
+        It "Binary split operator can works with freeform delimiter" {
+            $res = "a::b::c::d" -split "::"
+            $res.count | Should Be 4
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b"
+            $res[2] | Should Be "c"
+            $res[3] | Should Be "d"
+        }
+
+        It "Binary split operator can preserve delimiter" {
+            $res = "a1:b1:c1:d" -split "(1:)"
+            $res.count | Should Be 7
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "1:"
+            $res[2] | Should Be "b"
+            $res[3] | Should Be "1:"
+            $res[4] | Should Be "c"
+            $res[5] | Should Be "1:"
+            $res[6] | Should Be "d"
+
+            $res = "a1:b1:c1:d" -split "1(:)"
+            $res.count | Should Be 7
+            $res[0] | Should Be "a"
+            $res[1] | Should Be ":"
+            $res[2] | Should Be "b"
+            $res[3] | Should Be ":"
+            $res[4] | Should Be "c"
+            $res[5] | Should Be ":"
+            $res[6] | Should Be "d"
+        }
+
+        It "Binary split operator can be case-insensitive and case-sensitive" {
+            $res = "abcBd" -split "B"
+            $res.count | Should Be 3
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "c"
+            $res[2] | Should Be "d"
+
+            $res = "abcBd" -isplit "B"
+            $res.count | Should Be 3
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "c"
+            $res[2] | Should Be "d"
+
+            $res = "abcBd" -csplit "B"
+            $res.count | Should Be 2
+            $res[0] | Should Be "abc"
+            $res[1] | Should Be "d"
+
+            $res = "abcBd" -csplit "B", 0 , 'IgnoreCase'
+            $res.count | Should Be 3
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "c"
+            $res[2] | Should Be "d"
+        }
+
+        It "Binary split operator can works with script block" {
+            $res = "a::b::c::d" -split {$_ -eq "b" -or $_ -eq "C"}
+            $res.count | Should Be 3
+            $res[0] | Should Be "a::"
+            $res[1] | Should Be "::"
+            $res[2] | Should Be "::d"
+        }
+
+    }
+
+    Context "Binary split operator options" {
+        BeforeAll {
+            $testText = "a12a`nb34b`nc56c`nd78d"
```

Comment:
Multi-line tests should probably all test the three linefeed variants: `` `n ``, `` `r ``, and `` `r`n ``.

Reply:
Here we is full based on RegEx implementation in our code. We don't parse the string, we only pass it to RegEx as is.  Do we really want to test CoreFX implementation?

Reply:
True today and probably true forever, but if somebody decided they could be smarter than regex (maybe in limited scenarios), it's useful (if it's easy) to cover those additional scenarios.

Reply:
I see your point. I added tests for '`n' and '`r`n'. SingleLine option catch only '`n` so I believe no tests needed for '`r'.